plumbing/storer/reference.go

```diff
@@ -18,6 +18,7 @@ type ReferenceStorer interface {
 	SetReference(*plumbing.Reference) error
 	Reference(plumbing.ReferenceName) (*plumbing.Reference, error)
 	IterReferences() (ReferenceIter, error)
+	RemoveReference(plumbing.ReferenceName) error
```

Comment:
Being thread-safe or not is implementation-dependent. So far, none of our public implementations is thread-safe for any write operation.